detection-rules/brand_impersonation_sendgrid.yml

```
@@ -56,17 +74,44 @@ source: |
     or any(ml.nlu_classifier(beta.ocr(beta.message_screenshot()).text).intents,
            .name == "cred_theft" and .confidence == "high"
     )
+    // Infrastructure analysis for behavioral patterns
+    or (
+      any(body.links,
+          (
+            .href_url.path =~ "/ls/click"
+            or .href_url.path =~ "/wf/open"
+            or strings.contains(.href_url.query_params, "upn=")
+          )
+          and .href_url.domain.root_domain not in (
+            "sendgrid.com",
+            "twilio.com",
+            "mailchimp.com",
+            "constantcontact.com"
+          )
+      )
+      and (
+        any(body.links, length(.href_url.query_params) > 150)
+        or any(body.links,
+               .display_url.domain.root_domain in ("sendgrid.com", "twilio.com")
+        )
+        or body.current_thread.text =~ "(?i)(update.*billing|declined|api.*issue|misconfigured.*headers)"
+      )
+    )
   )
   
   // and the sender is not in org_domains or from sendgrid domains and passes auth
   and not (
     sender.email.domain.root_domain in $org_domains
     or (
-      sender.email.domain.root_domain in ("sendgrid.com", "sendgrid.net")
+      sender.email.domain.root_domain in (
+        "sendgrid.com",
+        "sendgrid.net",
+        "twilio.com"
```

```suggestion
        "twilio.com",
        "swoogo.com" // events planning software used by Twillio
```
https://platform.sublime.security/messages/4f675a3e9787138ec1a935d666541c238beb6c2a5e086b77a9cef3fdc9481670?preview_id=0198ee11-5e7f-7cc8-ad2d-99d4050aa1fa